vue/components/ui/molecules/animated-button-icon/animated-button-icon.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+    <div class="animated-button-icon">
+        <button-icon
+            class="button button-icon-front"
+            v-bind:icon="icon"
+            v-bind:color="color"
+            v-bind:size="size"
+            v-bind:disabled="disabled"
+            v-bind:class="{ visible: !clicked }"
+            v-on:click="onClick"
+        />
+        <button-icon
+            class="button button-icon-click"
+            v-bind:icon="clickIcon"
+            v-bind:color="color"
+            v-bind:size="size"
+            v-bind:disabled="true"
+            v-bind:class="{ visible: clicked }"
+        />
+    </div>
+</template>
+
+<style lang="scss" scoped>
+@import "css/variables.scss";
+
+.animated-button-icon {
+    display: inline-block;
+    position: relative;
+}
+
+.button {
+    opacity: 0;
+    pointer-events: none;
+    transition: opacity 0.125s linear;
+}
+
+.button.visible {
+    opacity: 1;
+}
+
+.button.button-icon-front.visible {
+    pointer-events: auto;
+}
+
+.button.button-icon-click {
+    left: 0px;
+    opacity: 0;
+    pointer-events: none;
+    position: absolute;
+    top: 0px;
+}
+
+.button.button-icon-click.visible {
+    opacity: 1;
+}

```

Comment:
Would you please refactor this to stop using the `button` class? My rationale is that the properties between the 2 buttons are always opposed to each other, and per the current way this is written that's not clear and takes a bit of juggling to realize what's going on.

The `.button.button-icon-front.visible` is a sympton of this, where you are overriding a property that shouldn't ever have been set in the first place.